[Data Quality] Update data quality detail issues table

SoniaSanzV opened the pull request:
Fixes https://github.com/elastic/kibana/issues/231483

## Summary
It updates the issues table following the mocks. Includes:
* Remove accordion table
* Change the icon to open the flyout and move it to the right of the table.
* Add a button to open the field in discovery.
* Split issue column to two columns: Field and type.
* Move toggle to a filter button
* Add filters for field and issue
* Changes the chat type and style and split the doc count and the chart into two columns to match the mocks style.



https://github.com/user-attachments/assets/d6e2786b-049c-40d9-93e6-b9dd330807f5

SoniaSanzV committed:
Update data quality detail issues table

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/329829)
* Commit: aaa93298edb5260d10ef868da53386d7f1830f6d

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/329829#0198a9fe-2205-4f10-81c1-30a9150eae63)

### History
* :green_heart: [Build #329780](https://buildkite.com/elastic/kibana-pull-request/builds/329780) succeeded a841bcae6428b3f02cda15bd0f2cd115fd6e5515

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329829","number":329829,"commit":"aaa93298edb5260d10ef868da53386d7f1830f6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329780","number":329780,"commit":"a841bcae6428b3f02cda15bd0f2cd115fd6e5515"}],"number":329829}
buildkite-pr-comment-->

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

SoniaSanzV committed:
Fix linter

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/329906)
* Commit: 48dbb29cb8013d9aa9825f836c6a8700c7f6f578

### Failed CI Steps
* [FTR Configs #124](https://buildkite.com/elastic/kibana-pull-request/builds/329906#0198accd-91a2-4b96-b780-b7252a22b7f0)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9a846293b5dd92351ddd03c7c7edecccecd375ff) | [after](https://github.com/elastic/kibana/commit/48dbb29cb8013d9aa9825f836c6a8700c7f6f578) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 818 | 823 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9a846293b5dd92351ddd03c7c7edecccecd375ff) | [after](https://github.com/elastic/kibana/commit/48dbb29cb8013d9aa9825f836c6a8700c7f6f578) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 423.6KB | 427.5KB | +3.9KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9a846293b5dd92351ddd03c7c7edecccecd375ff) | [after](https://github.com/elastic/kibana/commit/48dbb29cb8013d9aa9825f836c6a8700c7f6f578) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 24.2KB | 24.2KB | +9.0B |



### History
* :broken_heart: [Build #329864](https://buildkite.com/elastic/kibana-pull-request/builds/329864) failed 3d58b318d90690ea87f32592e6edb81277c64f74
* :broken_heart: [Build #329829](https://buildkite.com/elastic/kibana-pull-request/builds/329829) failed aaa93298edb5260d10ef868da53386d7f1830f6d
* :green_heart: [Build #329780](https://buildkite.com/elastic/kibana-pull-request/builds/329780) succeeded a841bcae6428b3f02cda15bd0f2cd115fd6e5515

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329906","number":329906,"commit":"48dbb29cb8013d9aa9825f836c6a8700c7f6f578"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329864","number":329864,"commit":"3d58b318d90690ea87f32592e6edb81277c64f74"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329829","number":329829,"commit":"aaa93298edb5260d10ef868da53386d7f1830f6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329780","number":329780,"commit":"a841bcae6428b3f02cda15bd0f2cd115fd6e5515"}],"number":329906}
buildkite-pr-comment-->

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

SoniaSanzV committed:
Address feedback

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

SoniaSanzV commented:
Hey @achyutjhunjhunwala, I've addressed your feedback in 20f6cf884a017bfd285da01889a82e5c6e5123ac. Also, as agreed with @MichaelMarcialis and @patpascal, I've modified the table changing the expand button back to the left and removed the discover icon link:

<img width="1268" height="237" alt="Screenshot 2025-08-19 at 14 26 49" src="https://github.com/user-attachments/assets/a49d1605-f467-4995-b6f1-411f3f5d32e5" />

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330773)
* Commit: 2b3743659254f59da75634cfe8f1e3c336bc4791

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/85556e777883bd5bd15f79313151dc80c4f57b08) | [after](https://github.com/elastic/kibana/commit/2b3743659254f59da75634cfe8f1e3c336bc4791) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 819 | 824 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/85556e777883bd5bd15f79313151dc80c4f57b08) | [after](https://github.com/elastic/kibana/commit/2b3743659254f59da75634cfe8f1e3c336bc4791) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 423.8KB | 426.7KB | +3.0KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/85556e777883bd5bd15f79313151dc80c4f57b08) | [after](https://github.com/elastic/kibana/commit/2b3743659254f59da75634cfe8f1e3c336bc4791) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 24.2KB | 24.2KB | +9.0B |



### History
* :green_heart: [Build #330378](https://buildkite.com/elastic/kibana-pull-request/builds/330378) succeeded 6c9751b0061e192e9d2268cbf0b2fb2a210311c1
* :broken_heart: [Build #330308](https://buildkite.com/elastic/kibana-pull-request/builds/330308) failed 8e76c408298c69a3bad0625d296f155394bc0387
* :yellow_heart: [Build #329906](https://buildkite.com/elastic/kibana-pull-request/builds/329906) was flaky 48dbb29cb8013d9aa9825f836c6a8700c7f6f578
* :broken_heart: [Build #329864](https://buildkite.com/elastic/kibana-pull-request/builds/329864) failed 3d58b318d90690ea87f32592e6edb81277c64f74
* :broken_heart: [Build #329829](https://buildkite.com/elastic/kibana-pull-request/builds/329829) failed aaa93298edb5260d10ef868da53386d7f1830f6d
* :green_heart: [Build #329780](https://buildkite.com/elastic/kibana-pull-request/builds/329780) succeeded a841bcae6428b3f02cda15bd0f2cd115fd6e5515

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330773","number":330773,"commit":"2b3743659254f59da75634cfe8f1e3c336bc4791"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330378","number":330378,"commit":"6c9751b0061e192e9d2268cbf0b2fb2a210311c1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330308","number":330308,"commit":"8e76c408298c69a3bad0625d296f155394bc0387"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329906","number":329906,"commit":"48dbb29cb8013d9aa9825f836c6a8700c7f6f578"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329864","number":329864,"commit":"3d58b318d90690ea87f32592e6edb81277c64f74"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329829","number":329829,"commit":"aaa93298edb5260d10ef868da53386d7f1830f6d"}],"number":330773}
buildkite-pr-comment-->

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/common/translations.ts:
```diff
@@ -437,6 +453,27 @@ export const documentIndexFailed = i18n.translate(
+    defaultMessage: 'Field ignored',
+  }
+);
+
```
can we include here something for when we show failedDocs?
<img width="1457" height="80" alt="image" src="https://github.com/user-attachments/assets/51b23edd-0460-4a58-8cc9-73eeac4be5d9" />
Or maybe use `Failed documents` or `Failed docs` if already exists

SoniaSanzV replied:
Sorry, I'm not sure what you mean. Do you mean to include a copy to show in the Field column instead of failedDocs as it is?

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/overview/quality_issues/columns.tsx:
```diff
@@ -79,36 +67,37 @@ export const getQualityIssuesColumns = ({
+      return <EuiText size="s">{countValue}</EuiText>;
+    },
+  },
+  {
```
Why do we add a new column here for the chart?
The problem with this is when you see the page in lower resolutions, the doc count and the chart are not together/aligned
<img width="1727" height="854" alt="image" src="https://github.com/user-attachments/assets/2ffc53c3-6469-4934-9748-7829e470046a" />

SoniaSanzV replied:
It was the way I found of align the columns in the way that Michael wanted, but I guess it's better to have the title align to the left and not lose that the chart and the count are together when resizing 

<img width="571" height="226" alt="Screenshot 2025-08-26 at 18 41 07" src="https://github.com/user-attachments/assets/38ecd9ca-5cc3-478b-bbe0-e0edd5f35532" />

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/overview/quality_issues/filters/degraded_fields_toggle_button.tsx:
can we rename this file to `current_quality_issues_toggle.tsx`?

yngrdyn commented on x-pack/platform/plugins/shared/dataset_quality/public/state_machines/dataset_quality_details_controller/state_machine.ts:
```diff
@@ -279,6 +279,15 @@ export const createPureDatasetQualityDetailsControllerStateMachine = (
                     },
+                    UPDATE_SELECTED_ISSUE_TYPES: {
+                      target:
+                        '#DatasetQualityDetailsController.initializing.dataStreamSettings.qualityIssues.dataStreamDegradedFields.fetchingDataStreamDegradedFields',
```
Why do we need to refetch degraded fields here?

SoniaSanzV replied:
We don't

yngrdyn reviewed (commented):
I finished a first round of review, left some comments
Looking good 🔥

MichaelMarcialis commented on x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality_details/overview/quality_issues/columns.tsx:
```diff
@@ -10,52 +10,40 @@ import { EuiButtonIcon, EuiText, formatNumber } from '@elastic/eui';
 }: {
   dateFormatter: FieldFormat;
   isLoading: boolean;
-  expandedQualityIssue?: {
```
Thanks for the ping! If we wish to proceed with the flyout overlaying the page contents, I believe the way @SoniaSanzV currently has it would be best, with the expand icon persisting while the flyout is open. This is because the overlay covers over the button, preventing the user from clicking it. Also, clicking anywhere outside an overlay modal will close the modal.

However, if there has been a change of heart and ya'll would prefer to use a push flyout, then I can see keeping the behavior of the expand icon changing when the flyout is open, as it would still be something the user could interact with.

patpascal replied:
Hi there! Given enough space, I'd recommend the flyout would be pushing rather than overlaying, as it currently does in Discover; this would imply that the icon would need to change to reflect the status.

SoniaSanzV replied:
I'll change the icon as part of this task, and we can discuss change the flyout back to push instead of overlay as a follow up!

SoniaSanzV committed:
Merge branch 'main' into data_quality/issues_table

SoniaSanzV committed:
Address feedback

SoniaSanzV committed:
Fix filter

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/333299)
* Commit: d9af053b28c7df1d69d1c636ab0492dcc3214c3a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/c0dbfacf55f19edf2bf9d9d5d401808dc5a5b69d) | [after](https://github.com/elastic/kibana/commit/d9af053b28c7df1d69d1c636ab0492dcc3214c3a) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 830 | 835 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c0dbfacf55f19edf2bf9d9d5d401808dc5a5b69d) | [after](https://github.com/elastic/kibana/commit/d9af053b28c7df1d69d1c636ab0492dcc3214c3a) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 431.9KB | 435.6KB | +3.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c0dbfacf55f19edf2bf9d9d5d401808dc5a5b69d) | [after](https://github.com/elastic/kibana/commit/d9af053b28c7df1d69d1c636ab0492dcc3214c3a) | diff |
| --- | --- | --- | --- |
| `datasetQuality` | 24.5KB | 24.5KB | +9.0B |



### History
* :broken_heart: [Build #333174](https://buildkite.com/elastic/kibana-pull-request/builds/333174) failed bb3e7dde22d60d30f1e7067227ff37df37659368
* :yellow_heart: [Build #331681](https://buildkite.com/elastic/kibana-pull-request/builds/331681) was flaky 07958d77fa603535ab98863863794aff9be751d3
* :green_heart: [Build #330773](https://buildkite.com/elastic/kibana-pull-request/builds/330773) succeeded 2b3743659254f59da75634cfe8f1e3c336bc4791
* :green_heart: [Build #330378](https://buildkite.com/elastic/kibana-pull-request/builds/330378) succeeded 6c9751b0061e192e9d2268cbf0b2fb2a210311c1
* :broken_heart: [Build #330308](https://buildkite.com/elastic/kibana-pull-request/builds/330308) failed 8e76c408298c69a3bad0625d296f155394bc0387
* :yellow_heart: [Build #329906](https://buildkite.com/elastic/kibana-pull-request/builds/329906) was flaky 48dbb29cb8013d9aa9825f836c6a8700c7f6f578

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333299","number":333299,"commit":"d9af053b28c7df1d69d1c636ab0492dcc3214c3a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333174","number":333174,"commit":"bb3e7dde22d60d30f1e7067227ff37df37659368"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331681","number":331681,"commit":"07958d77fa603535ab98863863794aff9be751d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330773","number":330773,"commit":"2b3743659254f59da75634cfe8f1e3c336bc4791"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330378","number":330378,"commit":"6c9751b0061e192e9d2268cbf0b2fb2a210311c1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330308","number":330308,"commit":"8e76c408298c69a3bad0625d296f155394bc0387"}],"number":333299}
buildkite-pr-comment-->

SoniaSanzV committed:
Remove saved issues types

yngrdyn reviewed (approved):
LGTM, good job 🚀
